### PR TITLE
Enable the e2e (integration) tests

### PR DESCRIPTION
**Pull request by enarha**

### CODE REVIEW

**Comment by adambkaplan** [comment]
/approve

Woot! e2e tests are now running as part of the `pull-tekton-results-integration-tests` suite!

```
+ go test -v -count=1 --tags=e2e /home/prow/go/src/github.com/tektoncd/results/test/e2e/...
=== RUN   TestTaskRun
    e2e_test.go:85: Created TaskRun hello
=== RUN   TestTaskRun/Result_ID
    e2e_test.go:91: Get: hello <nil>
    e2e_test.go:91: Get: hello <nil>
    e2e_test.go:96: Found Result: default/results/2e479200-1fc2-4996-b9d0-2dc8e[652](https://prow.tekton.dev/view/gs/tekton-prow/pr-logs/pull/tektoncd_results/287/pull-tekton-results-integration-tests/1604858029633703936#1:build-log.txt%3A652)0577
=== RUN   TestTaskRun/Run_Cleanup
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
    e2e_test.go:111: Get: hello, <nil>
--- PASS: TestTaskRun (23.14s)
    --- PASS: TestTaskRun/Result_ID (1.02s)
    --- PASS: TestTaskRun/Run_Cleanup (22.01s)
=== RUN   TestPipelineRun
    e2e_test.go:147: Found Result: default/results/54945bc9-584d-47cf-9b6f-118f83a4dbcf
--- PASS: TestPipelineRun (1.06s)
=== RUN   TestListResults
=== RUN   TestListResults/list_results_under_the_default_parent
    e2e_test.go:181: Found 2 results
=== RUN   TestListResults/list_results_across_parents
=== RUN   TestListResults/return_an_error_because_the_identity_isn't_authorized_to_access_all_namespaces
=== RUN   TestListResults/list_results_under_the_default_parent_using_the_identity_with_more_limited_access
    e2e_test.go:236: Found 2 results
--- PASS: TestListResults (1.99s)
    --- PASS: TestListResults/list_results_under_the_default_parent (0.60s)
    --- PASS: TestListResults/list_results_across_parents (0.80s)
    --- PASS: TestListResults/return_an_error_because_the_identity_isn't_authorized_to_access_all_namespaces (0.40s)
    --- PASS: TestListResults/list_results_under_the_default_parent_using_the_identity_with_more_limited_access (0.20s)
=== RUN   TestListRecords
=== RUN   TestListRecords/list_records_by_omitting_the_result_name
    e2e_test.go:254: Found 3 records
=== RUN   TestListRecords/list_records_by_omitting_the_parent_and_result_names
=== RUN   TestListRecords/return_an_error_because_the_identity_isn't_authorized_to_access_all_namespaces
=== RUN   TestListRecords/list_records_using_the_identity_with_more_limited_access
    e2e_test.go:309: Found 3 records
--- PASS: TestListRecords (1.00s)
    --- PASS: TestListRecords/list_records_by_omitting_the_result_name (0.20s)
    --- PASS: TestListRecords/list_records_by_omitting_the_parent_and_result_names (0.51s)
    --- PASS: TestListRecords/return_an_error_because_the_identity_isn't_authorized_to_access_all_namespaces (0.08s)
    --- PASS: TestListRecords/list_records_using_the_identity_with_more_limited_access (0.20s)
PASS
ok  	github.com/tektoncd/results/test/e2e	27.214s
```

**Comment by adambkaplan** [comment]
@enarha do we need to wait for https://github.com/tektoncd/plumbing/pull/1304 to merge, or can this go in now?

cc @afrittoli

**Comment by enarha** [comment]
I think we better get https://github.com/tektoncd/plumbing/pull/1304 in first, re-test with the new configuration, then merge. It's more controlled that way, so a change in `plumbing` do not break the running tests here.

**Comment by adambkaplan** [comment]
/hold

Per @enarha's comment, waiting for https://github.com/tektoncd/plumbing/pull/1304 to merge so tests run on kind-optimized notes.

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -279,7 +279,8 @@ func TestListRecords(t *testing.T) {
 			t.Fatal(err)
 		}
 
-		if diff := cmp.Diff(want.Records, got.Records, protocmp.Transform()); diff != "" {
+		// Compare only record names. Comparing records data is susceptable to race conditions.
+		if diff := cmp.Diff(recordNames(t, want.Records), recordNames(t, got.Records), protocmp.Transform()); diff != "" {
```
Comment (adambkaplan) [inline test/e2e/e2e_test.go:283]: @alan-ghelardi want to check with you if this is acceptable. I believe this is the intent of your test?

Comment (alan-ghelardi) [reply test/e2e/e2e_test.go:283]: I'm wondering whether this race condition is exactly what this [pull request](https://github.com/tektoncd/results/pull/298) is fixing. Anyhow, comparing only the record names looks good to me. After merging both prs, I'd like to write a test case to cover that bug.
/lgtm

Comment (enarha) [reply test/e2e/e2e_test.go:283]: IMO the race condition I'm observing is not related. Basically the e2e tests in CI take much longer (6+ mins compared to 2+ mins locally), so when the TaskRun and PipelineRun are called at the beginning of the e2e tests, then doing 2 ListRecords calls later was getting small differences in the data (mainly container state and timestamps) as things are still running. I was able to reproduce this locally by introducing 1-2 seconds delay between the 2 ListRecords calls and I was also able to eliminate the issue with introducing long enough delay after the TaskRun and PipelineRun calls and before the ListRecords calls.

Comment (alan-ghelardi) [reply test/e2e/e2e_test.go:283]: Makes sense.

**Review by khrm — COMMENTED** [review]
/kind cleanup
We need this label for release as discussed in the last WG call.

**Pull request merged**